score_analysis/showbias_plotting.py

```diff
@@ -9,8 +9,11 @@
 
 
 def plot_single_threshold(
-    bias_frame: BiasFrame, threshold: float, title: Optional[str] = None
-) -> plt:
+    bias_frame: BiasFrame,
+    threshold: float,
+    title: Optional[str] = None,
+    ax: Optional[plt.Axes] = None,
+) -> plt.Figure:
```

Comment:
Does the function need to return anything? It is always possible to obtain the current axes using `plt.gca()`.

Reply:
True but that would imply users would need to run an extra line (`plt.gca()`), right?

I am assuming here that the most common use case is that users do want to do something with the figure (e.g. save it to disk / a bucket or modify the figure further). What would be the advantage of not returning the figure?

Reply:
In my experience the default way of plotting is via `plt.plot()` and other functions. These all operate on the currently active figure. Unless you need to deal with multiple figures simultaneously, you don't care about the figure. You just write something like:
```
plt.figure(figsize=(6, 4))
plt.plot(x, y)
plt.xlabel("x")
plt.title("My cool graph")
plt.yscale("log")
plt.show()
```
This is how 90% of my plotting code looks like.

Reply:
Yes, same here. So running `plt.gca()` is actually usually not necessary. `plt.show()` or `plt.savefig()` will work just fine. In that case, it's indeed unnecessary to return the plot, removed it.